### ISSUE TITLE
Question about the encoder choice

### ISSUE THREAD

**Issue by its5Q** [OliBomby/Mapperatorinator#10]
Hi, saw you using Whisper as an audio encoder. I'm curious, why did you specifically choose Whisper? It was trained mostly on speech, perhaps with some background noise, so I would assume it wouldn't make for a good music encoder, but IIRC it was used successfully in Qwen2-Audio that works with speech.
There are some recent models designed specifically for music encoding, e.g. https://arxiv.org/pdf/2306.00107
Do you think replacing the Whisper backend could improve overall output quality?

btw, I have been interested in AI mapping for a while, are there any communities around AI mapping, or where you are the most active on? I'd love to chat with people about that stuff :)

**Comment by OliBomby** [comment]
I use Whisper because its an encoder-decoder model and it had better speed and accuracy than T5. Any encoder-decoder architecture works. That Whisper is usually trained on speech does not matter because I retrain it from scratch (random initialisation) on osu! music.

There are some people developing AI mapping in the osu!dreamer discord.

**Comment by its5Q** [comment]
Oh, I see, I thought you were using the model pretrained on speech-to-text and just using the encoder part of it, but now, as I can understand, you are training both the Whisper's encoder and decoder to generate the beatmap tokens, instead of text as in the original use-case of Whisper.
I feel like it could be beneficial to frankenstein a model with just an encoder taken from the paper I linked above, but the decoder left as-is from Whisper. I feel like using what should be a pretty good pretrain of a music encoder would work well in this case and could be further finetuned on osu! music specifically. That approach might yield better results for less compute, and I'm assuming we don't have that much compute.
Just bouncing some possibly worthwile ideas around

**Comment by OliBomby** [comment]
I'm skeptical that swapping the encoder for a pre-trained encoder for music understanding tasks would give immediate improvements. I think the Whisper encoder that I trained for the sole purpose of generating beatmaps has the best chance of extracting features that are useful for generating beatmaps.

Other pre-trained models might be useful for other use-cases, for example compressing audio so we can increase the context length with latent diffusion.

**Comment by OliBomby** [comment]
I might revisit your idea in a different way. I think a decoder-only architecture training from a pretrained multi-modal LLM might work and could increase parameter count with same compute.
